File: Monsoon Break/chat2.java
import java.util.*;

public class chat2 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int T = sc.nextInt();

        while (T-- > 0) {
            int N = sc.nextInt();
            String S = sc.next();
            int K = sc.nextInt();

            int left = 0, work = 0, ans = 0;

            for (int right = 0; right < N; right++) {
                if (S.charAt(right) == 'W') {
                    work++;
                }

                while (work > K) {
                    if (S.charAt(left) == 'W') {
                        work--;
                    }
                    left++;
                }

                ans = Math.max(ans, right - left + 1);
            }

            System.out.println(ans);
        }

        sc.close();
    }
}

File: Monsoon Break/chat2.py
T=int(input())

for _ in range(T):
    N=int(input())
    S=input().strip()
    K=int(input())

    left=0
    work=0
    ans=0

    for right in range(N):
        if S[right]=='W':
            work+=1

        while work>K:
            if S[left]=='W':
                work-=1
            left+=1

        ans=max(ans,right-left+1)

    print(ans)